app/src/main/java/com/swent/assos/ui/screens/Profil.kt

```diff
@@ -18,43 +22,136 @@ import androidx.compose.material3.TopAppBar
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import com.swent.assos.R
+import com.swent.assos.model.data.Association
+import com.swent.assos.model.navigation.Destinations
+import com.swent.assos.model.navigation.NavigationActions
 
 @Composable
-fun Profil() {
+fun Profil(navigationActions: NavigationActions) {
+
+  val firstname = "Maximilien"
+  val surname = "GRIDEL"
+
+  val completeName = "$firstname $surname"
+
   Scaffold(
       topBar = {
         TopAppBar(
-            title = { Text("Profil") },
+            title = { Text(text = completeName) },
             actions = {
               // Ajouter ici l'icône des paramètres
+
               Icon(
                   painterResource(id = R.drawable.settings),
                   contentDescription = "Map",
-                  modifier = Modifier.width(40.dp).height(40.dp),
+                  modifier =
+                      Modifier.width(40.dp).height(40.dp).clickable {
+                        navigationActions.navigateTo(destination = Destinations.SETTINGS)
+                      },
                   tint = MaterialTheme.colorScheme.onSurface)
             })
       }) { innerPadding ->
-        Column(modifier = Modifier.fillMaxSize().padding(innerPadding)) {
+        LazyColumn(
+            contentPadding = innerPadding,
+            modifier = Modifier.background(MaterialTheme.colorScheme.background).padding(16.dp)) {
+              item {
+                Text(text = "My associations", style = MaterialTheme.typography.titleLarge)
+                Spacer(modifier = Modifier.height(20.dp))
+              }
+
+              items(myAssociations) {
+                Card(
+                    modifier = Modifier.fillMaxWidth().padding(8.dp).height(50.dp),
+                    colors =
+                        CardDefaults.cardColors(
+                            containerColor = MaterialTheme.colorScheme.background)) {
+                      Text(text = it.acronym)
+                    }
+              }
 
-          // Card pour "My associations"
-          ProfileCard(title = "My associations", height = 100)
-          ProfileCard(title = "Associations I follow", height = 100)
-        }
+              item {
+                Text(text = "Associations followed", style = MaterialTheme.typography.titleLarge)
+                Spacer(modifier = Modifier.height(20.dp))
+              }
+
+              items(associationsFollowed) {
+                Card(
+                    modifier = Modifier.fillMaxWidth().padding(8.dp).height(50.dp),
+                    colors =
+                        CardDefaults.cardColors(
+                            containerColor = MaterialTheme.colorScheme.background)) {
+                      Text(text = it.acronym)
+                    }
+              }
+            }
       }
 }
```

Comment:
I believe it would be clearer if you separate your work into multiple sub composables.